junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -129,19 +132,18 @@ boolean hasException() {
 
     void notifyException(Exception exception) {
         this.exception = exception;
-        // Commented because of issue https://github.com/k3po/k3po/issues/391
-//        prepared.countDown();
-//        startable.countDown();
-//        finished.countDown();
-        if (testThread != null) {
-            if (testThread != Thread.currentThread())
-                testThread.interrupt();
-        }
+        interruptTestThread();
```

Comment:
If it is illegal to call `notifyException` with a `null` exception, then the following would be even simpler.
```
void notifyException(Exception exception) {
    this.exception = Objects.requireNonNull(exception, "exception");

    if (!currentThread().equals(testThread)) {
        testThread.interrupt();
    }
}
```